### PR TITLE
Trigger tests when undrafted

### PR DESCRIPTION
**Pull request by thibaultdvx**
Currently, functional tests are not triggered when a PR is in draft mode. However, mark it as ready for review doesn't trigger the tests and it is necessary to push a fake commit to launch them.

I propose to specify the events that will trigger tests in a PR, and especially add the `ready_for_review` event. Formerly, the events were the [default ones](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows), i.e. `opened`, `synchronize` and `reopened`.

To summarize, if this PR is accepted:
- unittests and linter tests would always be run (even in draft mode),
- functional tests would not run in draft mode, but mark a PR as ready for review will trigger them.

### CODE REVIEW

**Review by NicolasGensollen — APPROVED** [review]
Good idea @thibaultdvx ! 👍 
LGTM !
